gdal/frmts/tiledb/tiledbdataset.cpp

```diff
@@ -1376,6 +1460,76 @@ TileDBDataset* TileDBDataset::CreateLL( const char *pszFilename,
         
         poDS->m_schema->set_domain( domain ).set_order({{TILEDB_ROW_MAJOR, TILEDB_ROW_MAJOR}});;
 
+        // register additional attributes to the pixel value, these will be
+        // be reported as subdatasets on future reads
+        poDS->papszAttributes = CSLFetchNameValueMultiple(
+                                                    papszOptions, "TILEDB_ATTRIBUTE" );
+
+        for( int i = 0;
+            poDS->papszAttributes != nullptr &&
+            poDS->papszAttributes[i] != nullptr;
+            i++ )
+        {
+            // modeling additional attributes as subdatasets
+            poDS->bHasSubDatasets = true;
+            // check each attribute is a GDAL source
+            std::unique_ptr<GDALDataset> poAttrDS(
```

Comment:
using GDALDataset::Open() will save the (GDALDataset*) cast

Reply:
:+1: